R/parasite_parameters.R

```diff
@@ -0,0 +1,233 @@
+#' * id0 - scale parameter; default = 27.52 (falc. = 1.577533)
+#' * kd - shape parameter; default = 2.403 (falc. = 0.476614)
+#' 
+#' immunity boost grace period changes
+#' * uc - period in which clinical immunity cannot be boosted; default = 7.85 (falc. = 6.06349)
+#' * ud - period in which immunity to detectability cannot be boosted; default = 19.77 (falc. = 9.44512)
+#' 
+#' immunity boost grace periods (pre-erythrocytic/blood and severe immunity not modelled)
+#' * ub - period in which pre-erythrocytic immunity cannot be boosted (removed)
+#' * uv - period in which severe immunity cannot be boosted (removed)
+#' 
+#' infectivity towards mosquitoes changes
+#' * cd - infectivity of clinically diseased humans towards mosquitoes; default = 0.8 (falc. = 0.068)
+#' * cu - infectivity of sub-patent infection; default = 0.035 (falc. = 0.0062)
+#' * ct - infectivity of treated infection; default = 0.4 (falc. = 0.022)
+#' * gamma1 - parameter for infectivity of asymptomatic humans (removed: asymptomatic infectivity now a parameter: see "ca")
+#' * ca - infectivity of asymptomatic infection; default = 0.1 (new: replaces age-dependent parameterisation)
+#' 
+#' unique biting rate changes
+#' * sigma_squared - heterogeneity parameter; default = 1.29 (falc. = 1.67)
+#' 
+#' incubation period changed
+#' * de - Duration of the human latent period of infection; default = 10 (falc. = 12)
+#' * delay_gam - Lag from parasites to infectious gametocytes; default = 0 (falc. = 12.5)
+#' * dem - Extrinsic incubation period in mosquito population model; default = 8.4 (falc. = 10)
+#' 
+#' human mortality parameter changes
+#' * pcm - new-born clinical immunity relative to mother's; default = 0.421 (falc. = 0.774368)
+#' * pvm - new-born severe immunity relative to mother's (removed: severe immunity not modelled)
+#' 
+#' initial immunity values changes
+#' * init_ivm - the immunity from severe disease at birth (removed: severe immunity no longer modelled)
+#' * init_ib - the initial pre-erythrocitic immunity (removed: pre-erythrocytic/blood immunity no longer modelled)
+#' * init_iva - the initial acquired immunity from severe disease (removed: severe immunity no longer modelled)
+#' * init_idm - the initial maternal immunity to detectability: default = 0 (new: now modelled)
+#'  
+#'  hypnozoite parameters introduced
+#' * f - relapse rate: default = 0.024 (new)
+#' * gammal - clearance rate: default = 0.0026 (new)
+#'  
+#'  miscellaneous parameter changes
+#' * parasite - Plasmodium species; default = "falciparum"
+#' 
+#' 
+#' @param parameters the model parameters
+#' @param parasite parasite species name: "falciparum" or "vivax"
+#' @export
+set_parasite <- function(parameters, parasite) {
+  if (length(parasite) != 1) {
+    stop('Only one parasite species can be selected')
+  }
+  if (!parasite %in% c("vivax","falciparum")){
+    stop('Parasite species must be "vivax" or "falciparum"')
+  }
+  if (parasite == "falciparum"){
+    return(parameters)
```

Comment:
does this method work if I do this?

```r
parameters <- set_parasite(parameters, 'vivax')
parameters <- set_parasite(parameters, 'falciparum')
```

I think we should either restrict this function to only go from falciparum -> vivax by renaming it `set_vivax`, or support going from vivax parameters to falciparum ones (probably less likely).

Reply:
I've completely reshifted the code, using a csv file for the parameters, and only loading the appropriate set in get_parameters. I think it saves a lot of space and faff.